scripts/collection_from_items.py

```diff
@@ -14,28 +14,48 @@
 
 
 def main() -> None:
+    # pylint: disable-msg=too-many-locals
     parser = argparse.ArgumentParser()
     parser.add_argument("--uri", dest="uri", help="s3 path to items and collection.json write location", required=True)
     parser.add_argument("--collection-id", dest="collection_id", help="Collection ID", required=True)
     parser.add_argument("--title", dest="title", help="Collection title", required=True)
     parser.add_argument("--description", dest="description", help="Collection description", required=True)
-    parser.add_argument("--producer", dest="producer", help="Imagery producer", required=True)
-    parser.add_argument("--licensor", dest="licensor", help="Imagery licensor")
-    parser.add_argument("--licensor-list", dest="licensor_list", help="Imagery licensor list")
+    parser.add_argument(
+        "--producer",
+        dest="producer",
+        help="Imagery producer. Ignored if --producer-list passed with a semicolon delimited list.",
+    )
+    parser.add_argument("--producer-list", dest="producer_list", help="Semicolon delimited list of imagery producers")
+    parser.add_argument(
+        "--licensor",
+        dest="licensor",
+        help="Imagery licensor. Ignored if --licensor-list passed with a semicolon delimited list.",
+    )
+    parser.add_argument("--licensor-list", dest="licensor_list", help="Semicolon delimited list of imagery licensors")
     parser.add_argument(
         "--concurrency", dest="concurrency", help="The number of files to limit concurrent reads", required=True, type=int
     )
 
     arguments = parser.parse_args()
     uri = arguments.uri
-    licensors: List[str] = parse_list(arguments.licensor_list)
-    if len(licensors) <= 1:
-        licensors = [arguments.licensor]
+
+    producers: List[str] = []
```

Comment:
I wonder if turning this into a function would be nicer?

```
def parse_list_(item_list: Optional[str], item_single: Optional[str]) -> List[str]:
 output = []
 if item_list:
￼  output.extend(parse_list(item_list)
￼elif item_single:
￼  output.append(item_single)
return output


for producer in parse_list(args.producer_list, args.producer):
  providers.append({ "name":... })
for licensor in parse_list(args.licensor_list, args.licensor):
  providers.append({ "name":... })
```

I also wonder if it would just be easier to allow `licensor` to be a `;` separated list

```
--licensor "Something;Something Else"
--licensor "Something"
--licensor '["Something","Something Else"]'
```

Rather than having two parameters? I guess the question would be how would that work in Argo?

Reply:
When I did the `producers` I ran into an issue with Argo not being able to wrap the list into `[ ... ]`

Reply:
Have refactored argument parsing into a helper function.

Changing to single argument would involve adding conditional logic to Argo workflow template and/or not having a dropdown in the Argo GUI, so would be best done via another PR.